editor/store/selectors.js

```diff
@@ -301,6 +301,44 @@ export function isEditedPostEmpty( state ) {
 	);
 }
 
+/**
+ * Returns true if the post can be autosaved, or false otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {boolean}       Whether the post can be autosaved
+ */
+export function isPostAutosavable( state ) {
+	// If the post is autosaving, it is not autosavable.
+	if ( state.currentlyAutosaving ) {
+		return false;
+	}
+
+	// If we don't already have an autosave, the post is autosavable.
+	if ( ! hasAutosave( state ) ) {
+		return true;
+	}
+
+	const title = getEditedPostAttribute( state, 'title' );
+	const excerpt = getEditedPostExcerpt( state );
+	const content = getEditedPostContent( state );
+	const autosave = state.editor.present.autosave;
+
+	// If the title, excerpt or content has changed, the post is autosavable.
+	if (
+		( autosave.title && title !== autosave.title.raw ) ||
+		( autosave.excerpt && excerpt !== autosave.excerpt.raw ) ||
```

Comment:
It's not clear to me when `autosave` would have or not have these properties. For example, if `getEditedPostExcerpt` returned a non-empty value but `autosave.excerpt` were empty, this would not return `true`. Is that expected? Unit tests should clarify these expectations.

Reply:
I think autosave will always have the excerpt and title properties - they can't be empty, they can only be an empty string which will evaluate correctly. The reason for this is that the api always returns all fields, so excerpt is always returned even if "empty". autosave represents the last stored autosave so we can be confident an excerpt will always be set.  I'm going to remove the checks for now since they are confusing and I don't think they are needed

Reply:
fixed in https://github.com/WordPress/gutenberg/pull/6257/commits/35f41ee3c7363492e61ddbc7181955850376fd20